### PR TITLE
feat(events)!: add config.merge option

### PR DESCRIPTION
**Pull request by swiatekm-sumo**
##### Description

Logs and metrics metadata config has `merge` and `override` properties to directly modify otel config. Make this so for events as well. This is a breaking change, as events already had `override`, but the meaning of it was identical to `merge`. We can handle this via an automatic migration.

Migration PR: https://github.com/SumoLogic/sumologic-kubernetes-tools/pull/389

---

##### Checklist

- [x] Changelog updated

### CODE REVIEW

**Comment by sumo-drosiek** [comment]
Please remove tmpcharts

File: docs/opentelemetry-collector.md [hunk 1]
```diff
@@ -598,9 +598,15 @@ options under `otelevents` in [values.yaml](/deploy/helm/sumologic/values.yaml)
...
 
 If the configuration options present under the `otelevents` key aren't sufficient for your needs, you can override
-the OT configuration directly. Be aware that doing this isn't subject to normal backwards-compatibility guarantees offered by
-this chart, so you'll need to be more careful during upgrades. The exact OT configuration emitted by the chart may change
-even in minor releases.
+the OT configuration directly.
 
-In order to override the configuration, use the `otelevents.config.override` key. This key takes a yaml object, whose
-value is merged with the configuration generated by the Chart.
+The `metadata.metrics.config.merge` key can be used to provide configuration that will be merged
```
Comment (sumo-drosiek) [inline docs/opentelemetry-collector.md:603]:
```suggestion
The `otelevents.config.merge` key can be used to provide configuration that will be merged
```

File: docs/opentelemetry-collector.md [hunk 2]
```diff
@@ -598,9 +598,15 @@ options under `otelevents` in [values.yaml](/deploy/helm/sumologic/values.yaml)
...
+with the Helm Chart's default configuration. It should be noted that this field is not subject to
+normal backwards compatibility guarantees, the default configuration can change even in minor
+versions while preserving the same end-to-end behaviour. Use of this field is discouraged - ideally
+the necessary customizations should be able to be achieved without touching the otel configuration
+directly. Please open an issue if your use case requires the use of this field.
+
+The `metadata.metrics.config.override` key can be used to provide configuration that will be completely
```
Comment (sumo-drosiek) [inline docs/opentelemetry-collector.md:610]:
```suggestion
The `otelevents.config.override` key can be used to provide configuration that will be completely
```

**Comment by swiatekm-sumo** [comment]
> Please remove tmpcharts

Whoops, done. I also added the directory to .gitignore.

File: .gitignore [hunk 3]
```diff
@@ -9,6 +9,8 @@
 # Helm chart dependencies
 deploy/helm/sumologic/charts
 deploy/helm/sumologic/charts/*
+deploy/helm/sumologic/tmpcharts
+deploy/helm/sumologic/tmpcharts/*
```
Comment (sumo-drosiek) [inline .gitignore:13]: Not sure. Why do we do `*` in addition to directory path?

**Review by sumo-drosiek — APPROVED** [review]
one nit regarding .gitignore

File: .gitignore (hunk 3 above)
Comment (swiatekm-sumo) [reply .gitignore:13]: I don't think it's needed actually. `.gitignore` docs say:

> The pattern foo/ will match a directory foo and paths underneath it, but will not match a regular file or a symbolic link foo (this is consistent with the way how pathspec works in general in Git)

So I'll remove the wildcards.

**Pull request merged**